### PR TITLE
[ORCA-189] Adds support for FASTQ and FASTQ-GZ files

### PR DESCRIPTION
**Pull request by BWMac**
This PR follows the example of #28 to add support for `.fastq`, `.fastq.gz`, `.fq`, and `.fq.gz` files to the package under one new `FASTQ` file type. This initial feature add includes the `FileExtensionTest` and the `MD5ChecksumTest`. It was successfully tested using the following commands:
```
dcqc qc-file -t FASTQ -m '{"md5_checksum": "14758f1afd44c09b7992073ccf00b43d"}' ./tests/data/example.fastq
```
Output:
```
{
  "type": "FastqSuite",
  "target": {
    "type": "Target",
    "id": null,
    "files": [
      {
        "url": "tests/data/example.fastq",
        "metadata": {
          "md5_checksum": "14758f1afd44c09b7992073ccf00b43d"
        },
        "type": "FASTQ",
        "name": "example.fastq",
        "local_path": "/var/folders/sr/3g4hnkfd4ld306tty7kqf1rr0000gr/T/dcqc-staged-qhtlwh_r/example.fastq"
      }
    ]
  },
  "suite_status": {
    "required_tests": [
      "FileExtensionTest",
      "Md5ChecksumTest"
    ],
    "skipped_tests": [],
    "status": "passed"
  },
  "tests": [
    {
      "type": "FileExtensionTest",
      "tier": 1,
      "is_external_test": false,
      "status": "passed"
    },
    {
      "type": "Md5ChecksumTest",
      "tier": 1,
      "is_external_test": false,
      "status": "passed"
    }
  ]
}
```
And...
```
dcqc qc-file -t FASTQ -m '{"md5_checksum": "14758f1afd44c09b7992073ccf00b43d"}' ./tests/data/example.fastq.gz
```
Output:
```
{
  "type": "FastqSuite",
  "target": {
    "type": "Target",
    "id": null,
    "files": [
      {
        "url": "tests/data/example.fastq.gz",
        "metadata": {
          "md5_checksum": "14758f1afd44c09b7992073ccf00b43d"
        },
        "type": "FASTQ",
        "name": "example.fastq.gz",
        "local_path": "/var/folders/sr/3g4hnkfd4ld306tty7kqf1rr0000gr/T/dcqc-staged-687p4xrg/example.fastq.gz"
      }
    ]
  },
  "suite_status": {
    "required_tests": [
      "FileExtensionTest",
      "Md5ChecksumTest"
    ],
    "skipped_tests": [],
    "status": "passed"
  },
  "tests": [
    {
      "type": "FileExtensionTest",
      "tier": 1,
      "is_external_test": false,
      "status": "passed"
    },
    {
      "type": "Md5ChecksumTest",
      "tier": 1,
      "is_external_test": false,
      "status": "passed"
    }
  ]
}
```

Note: The example `.fastq` and `.fastq.gz` files included in this PR are NOT real files. They are renamed TXT files due to the prohibitive size of these files. We will need to develop a solution for testing this package on large files in the future.

### CODE REVIEW

File: src/dcqc/file.py [hunk 1]
```diff
@@ -109,6 +109,8 @@ def get_file_type(cls, file_type: str) -> FileType:
 FileType("OME-TIFF", (".ome.tif", ".ome.tiff"), "format_3727")
 FileType("TSV", (".tsv"), "format_3475")
 FileType("BAM", (".bam"), "format_2572")
+FileType("FASTQ", (".fastq"), "format_1930")
```
Comment (thomasyu888) [inline src/dcqc/file.py:112]: Ohh, my apologies, I see what you meant by the OME-TIFF example now.

Were you thinking of this originally?

```
FileType("FASTQ", (".fastq", ".fastq.gz")...
```

Comment (BWMac) [reply src/dcqc/file.py:112]: Yeah, that's what I was initially thinking. That way might make more sense, but we would need to handle the zipped case eventually (unzip to run some tests) while also comparing the correct MD5 Checksum (zipped vs. unzipped) to the provided value. I.e. if someone provides the zipped MD5 sum we would need to run that check before unzipping the file.

Comment (thomasyu888) [reply src/dcqc/file.py:112]: Makes sense to me.  Let's make change and we can open up a ticket to deal with zipped files. Im sure we will run into this issue with other larger file formats (.tar.gz, gz, etc)

Can you confirm that to the md5checksum test works with an actual gunzipped file?

Comment (BWMac) [reply src/dcqc/file.py]: I made the changes discussed in this thread in the latest commit.

File: tests/data/example.fastq.gz [hunk 2]
```diff
@@ -0,0 +1 @@
+foobar
```
Comment (thomasyu888) [inline tests/data/example.fastq.gz:1]: TODO: create ticket to support compressed file formats in a formal way.

File: src/dcqc/file.py [hunk 3]
```diff
@@ -109,6 +109,7 @@ def get_file_type(cls, file_type: str) -> FileType:
 FileType("OME-TIFF", (".ome.tif", ".ome.tiff"), "format_3727")
 FileType("TSV", (".tsv"), "format_3475")
 FileType("BAM", (".bam"), "format_2572")
+FileType("FASTQ", (".fastq", ".fastq.gz"), "format_1930")
```
Comment (thomasyu888) [inline src/dcqc/file.py:112]: @BWMac Hold this for a second, but I think we have `fq` and `fq.gz` as well.

Comment (BWMac) [reply src/dcqc/file.py:112]: Looks like [BioPortal](https://bioportal.bioontology.org/ontologies/EDAM?p=classes&conceptid=format_1930) treats both extensions (`fq` and `fastq`) the same format too, so we could just add  `fq`and `fq.gz` to the tuple here.

Comment (BWMac) [reply src/dcqc/file.py]: added and tested successfully.

**Review by thomasyu888 — APPROVED** [review]
💯 LGTM! merge away!

**Pull request merged**